compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -10287,6 +10356,98 @@ private STNode parseLetClause(boolean isRhsExpr) {
         return STNodeFactory.createLetClauseNode(letKeyword, letVarDeclarations);
     }
 
+    /**
+     * Parse order by clause.
+     * <p>
+     * <code>order-by-clause := order by expression (ascending | descending)? (expression (ascending | descending)?)*
```

Comment:
this looks more cleaner to me. order-by-clause := order by (expression (ascending | descending)?)+
WDYT?